src/Amadevus.RecordGenerator.Generators/RecordPartialGenerator.cs

```diff
@@ -122,20 +122,58 @@ SyntaxToken GetMutatorIdentifier()
 
         public static MemberDeclarationSyntax GenerateToString(RecordDescriptor descriptor)
         {
-            var properties =
-                from e in descriptor.Entries
-                select AnonymousObjectMemberDeclarator(IdentifierName(e.Identifier));
-            return
-                MethodDeclaration(
-                    PredefinedType(Token(SyntaxKind.StringKeyword)),
-                    Names.ToString)
-                .AddModifiers(SyntaxKind.PublicKeyword, SyntaxKind.OverrideKeyword)
-                .WithExpressionBody(
-                    InvocationExpression(
-                        MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
-                            AnonymousObjectCreationExpression()
-                                .AddInitializers(properties.ToArray()),
-                            IdentifierName(Names.ToString))));
+            ExpressionSyntax expression;
+
+            if (descriptor.Entries.Length == 1)
+            {
+                var name = descriptor.Entries[0].Identifier;
+                var start = "{{ " + name + " = ";
+                var end = " }}";
+
+                expression = InterpolatedStringExpression(
```

Comment:
I think this'll be much easier as a concatenation, for both generating code, and the generated code.

```csharp
public override string ToString() => "{ Prop = " + Prop + " }";
```

Reply:
Yep, fair shout, looks much simpler in the SyntaxTree format